File: java/playwright/src/test/java/dev/selenium/request/ModifyPostTest.java
package dev.selenium.request;

import com.microsoft.playwright.Route;
import dev.selenium.TestBase;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class ModifyPostTest extends TestBase {

  @Test
  public void modifyPostBody() {
    page.navigate("https://httpbin.org/forms/post");
    page.fill("input[name=custemail]", "real@example.com");

    context.route(
        "**/post",
        route -> {
          String overrideBody = "custname=&custtel=&custemail=fake@example.com&delivery=&comments=";

          route.resume(new Route.ResumeOptions().setPostData(overrideBody));
        });
    page.click("button");

    String bodyText = page.innerText("body").replaceAll("\\s+", "");
    String expected = "\"custemail\":\"fake@example.com\"";
    Assertions.assertTrue(bodyText.contains(expected));
  }
}

File: dotnet/Playwright/Request/ModifyPostTest.cs
using System.Text;
using Microsoft.Playwright;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Playwright.Request;

[TestClass]
public class ModifyPostTest : TestBase
{
    [TestMethod]
    public async Task ModifyPostBody()
    {
        await Page!.GotoAsync("https://httpbin.org/forms/post");
        await Page.FillAsync("input[name=custemail]", "real@example.com");

        await Page.RouteAsync("**/post", async route =>
        {
            var overrideBody = "custname=&custtel=&custemail=fake@example.com&delivery=&comments=";
            await route.ContinueAsync(new RouteContinueOptions { PostData = Encoding.UTF8.GetBytes(overrideBody) });
        });

        await Page.ClickAsync("button");

        var expected = "\"custemail\": \"fake@example.com\"";
        var bodyText = await Page.TextContentAsync("body");
        Assert.IsTrue(bodyText!.Contains(expected));
    }
}